### PR TITLE
Adding PID QC at async time

### PR DESCRIPTION
**Pull request by chiarazampolli**
@catalinristea @noferini @knopers8 @davidrohr 
We would like to have this feature for the next apass4 of the pilot beam.

### CODE REVIEW

**Comment by noferini** [comment]
Hi Chiara,
many thanks!
Adding also @njacazio and @ercolessi to keep them informed.
Cheers
Francesco

File: DATA/common/setenv.sh [hunk 1]
```diff
@@ -160,6 +162,17 @@ has_matching_qc()
   has_detector_matching $1 && [[ $WORKFLOW_DETECTORS_QC =~ (^|,)"$1"(,|$) ]]
 }
 
+has_pid_qc()
+{
+    PIDDETECTORS=$(echo $1 | tr "-" "\n")
+    for PIDDETECTOR in $PIDDETECTORS;
+    do
+	echo PIDDETECTOR=$PIDDETECTOR 1>&2
+	! has_detector_qc $PIDDETECTOR ]] && return 1
```
Comment (martenole) [inline DATA/common/setenv.sh:171]: I assume that you have tested this and it works. But don't you need a matching `[[` for the `]]`?

Comment (chiarazampolli) [reply DATA/common/setenv.sh:171]: I did test it, I guess I made a typo later on (I was doing several tests yesterday in parallel). I will fix it. 
Thanks for spotting!

**Comment by chiarazampolli** [comment]
If WORKFLOW_DETECTORS contains TOF and FT0, this task is added to the QC json:

```
      "TaskFT0TOF": {
        "active": "true",
        "className": "o2::quality_control_modules::pid::TaskFT0TOF",
        "moduleName": "QcTOF",
        "detectorName": "TOF",
        "cycleDurationSeconds": "10",
        "maxNumberCycles": "-1",
        "dataSource": {
          "type": "direct",
          "query_comment": "checking every matched track",
          "query": "matchITSTPCTOF:TOF/MTC_ITSTPC/0;trackITSTPC:GLO/TPCITS/0;trackITSTPCABREFS:GLO/TPCITSAB_REFS/0;trackITSTPCABCLID:GLO/TPCITSAB_CLID/0;tofcluster:TOF/CLUSTERS/0;trackTPC:TPC/TRACKS/0;trackTPCClRefs:TPC/CLUSREFS/0;recpoints:FT0/RECPOINTS/0"
        },
        "taskParameters": {
          "GID": "ITS-TPC,TPC,ITS-TPC-TOF",
          "verbose": "false",
          "minPtCut": "0.3f",
          "etaCut": "0.8f",
          "minNTPCClustersCut": "60",
          "minDCACut": "100.f",
          "minDCACutY": "10.f",
          "grpFileName": "o2sim_grp.root",
          "geomFileName": "o2sim_geometry.root"
        },
        "": "For debugging, path to the file where to save. If empty or missing it won't save."
      }
```

File: DATA/common/setenv.sh (hunk 1 above)
Comment (chiarazampolli) [reply DATA/common/setenv.sh]: Fixed! Thanks again!

**Comment by chiarazampolli** [comment]
[20220519-093830-1290255-17263--ITS-MFT-TPC-FT0-MID-FV0-ITSTPC-ITSTPCTOF-PRIMVTX.json.txt](https://github.com/AliceO2Group/O2DPG/files/8726870/20220519-093830-1290255-17263--ITS-MFT-TPC-FT0-MID-FV0-ITSTPC-ITSTPCTOF-PRIMVTX.json.txt)
[20220519-093804-1290122-8514--ITS-MFT-TPC-TOF-MID-FV0-ITSTPC-ITSTPCTOF-PRIMVTX.json.txt](https://github.com/AliceO2Group/O2DPG/files/8726872/20220519-093804-1290122-8514--ITS-MFT-TPC-TOF-MID-FV0-ITSTPC-ITSTPCTOF-PRIMVTX.json.txt)
[20220519-093619-1289946-12000--ITS-MFT-TPC-TOF-FT0-MID-FV0-ITSTPC-ITSTPCTOF-PRIMVTX-pidFT0-TOF.json.txt](https://github.com/AliceO2Group/O2DPG/files/8726873/20220519-093619-1289946-12000--ITS-MFT-TPC-TOF-FT0-MID-FV0-ITSTPC-ITSTPCTOF-PRIMVTX-pidFT0-TOF.json.txt)

Attached the json with the different WORKFLOW_DETECTORS (with TOF and FT0, with no FT0, with no TOF). Extension had to be changed to "txt" to attach them here.

**Comment by chiarazampolli** [comment]
Hello @davidrohr , 

I have one remark on this PR: for now I only check "has_detector_qc" which checks:
```
has_detector $1 && [[ $WORKFLOW_DETECTORS_QC =~ (^|,)"$1"(,|$) ]]
```
In fact for TOF one needs matching to ITSTPC tracks. But this is specific of TOF, so I wonder if we should have a special case for TOF to check the matching etc. What do you think?

Cheers,

Chiara

**Comment by davidrohr** [comment]
if that QC needs the matching, and thus also ITS and TPC reco, I would check that it is present and disable it otherwise.

**Comment by chiarazampolli** [comment]
TOF-specific checks added. @davidrohr , do you think we can merge, or do you have further comments? It would be good to have this for the upcoming apass4, if you agree.

**Pull request merged**